src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -398,6 +398,52 @@ class %s extends {$options['class']}
         return $code;
     }
 
+    private function isPreloadable(string $class): ?bool
+    {
+        static $cache = [];
+        if (array_key_exists($class, $cache)) {
+            return $cache[$class];
+        }
+        $cache[$class] = true; // prevent recursion
+
+        if (!class_exists($class) && !interface_exists($class, false) && !trait_exists($class, false)) {
```

Comment:
Why did we checked for `interface_exists` and `trait_exists`?

Shouldn't we preload only classes? If a trait or an interface is used by a class, then it will be preloaded via the class that use it right?

Reply:
in theory you're right I think, in practice the worst that can happen is these checks being dead code
not having to think about this is why this code is here :)